Firestore/Source/API/FIRFirestore.mm

```diff
@@ -49,9 +49,11 @@
 
 extern "C" NSString *const FIRFirestoreErrorDomain = @"FIRFirestoreErrorDomain";
 
-@interface FIRFirestore ()
+@interface FIRFirestore () {
+  /** The actual owned DatabaseId instance is allocated in FIRFirestore. */
+  firebase::firestore::model::DatabaseId _databaseIDAlloc;
```

Comment:
Why "Alloc"? Why not just `_databaseID`?

Reply:
_databaseID is already claimed as a property in header for public usage. So this internal one is named differently.

Reply:
But the public databaseID property is this field.

Declaring the property exists in the header will do the following:
  * declare the instance variable if it doesn't already exist
  * declare the getter if it doesn't already exist
  * (if readwrite) declare the setter if it doesn't already exist

By supplying our own instance variable and getter on a readonly property, nothing is synthesized but callers can still access the property.

There's no need for a separate variable.

Reply:
Fixed. I was thinking it is confusing to use the same databaseID, where the public interface suggests to be pointer to DatabaseId while the implementation is actually C++ DatabaseId value. I guess this is practical in objective-c in particular when mixed with C++ type here.